varcode/effects.py

```diff
@@ -382,13 +395,42 @@ def __init__(
         self.aa_mutation_end_offset = aa_mutation_end_offset
         self.aa_ref = aa_ref
 
-    modifies_protein_sequence = True
+    @property
+    def modifies_protein_sequence(self):
+        return True
+
+class StartLoss(NonsilentCodingMutation):
+    """
+    When a start codon is lost it's difficult to determine if there is
+    an alternative Kozak consensus sequence (either before or after the
+    original) from which an alternative start codon can be inferred.
+    """
+    def __init__(
+            self,
+            variant,
+            transcript):
+        NonsilentCodingMutation.__init__(
+            self,
+            variant=variant,
+            transcript=transcript,
+            aa_mutation_start_offset=0,
+            aa_mutation_end_offset=1,
+            aa_ref=transcript.protein_sequence[0])
+
+    @property
+    def mutant_protein_sequence(self):
+        # TODO: scan downstream in the cDNA sequence to predict an alternate
+        # start codon using a Kozak sequence template
+        return None
+
+    @memoized_property
+    def short_description(self):
+        return "p.%s1? (start-loss)" % (self.transcript.protein_sequence[0],)
 
-class BaseSubstitution(NonsilentCodingMutation):
+class KnownAminoAcidChange(NonsilentCodingMutation):
```

Comment:
Does the `Known` part imply anything here? Why not just `AminoAcidChange`? This is everything that has `modifies_protein_sequence == True`?